### PR TITLE
[ssh_config] Add support for ControlMaster

### PR DESCRIPTION
**Pull request by mering**
##### SUMMARY
Add support for [`ControlMaster`](https://man.openbsd.org/ssh_config.5#ControlMaster) (with [`ControlPath`](https://man.openbsd.org/ssh_config.5#ControlPath), [`ControlPersist`](https://man.openbsd.org/ssh_config.5#ControlPersist)) SSH config options.

Second commit also converts `bool` to `str` correctly for options which allow `yes` / `no` in addition to other values as YAML `yes` / `no` is parsed as `True` / `False` but `ssh_config` needs `"yes"` / `"no"`.

##### ISSUE TYPE
- Feature Pull Request

##### COMPONENT NAME
ssh_config

##### ADDITIONAL INFORMATION
_None_

### CODE REVIEW

File: plugins/modules/ssh_config.py [hunk 1]
```diff
@@ -108,6 +108,19 @@
       - Sets the C(HostKeyAlgorithms) option.
     type: str
     version_added: 6.1.0
+  controlmaster:
+    description:
+      - Sets the C(ControlMaster) option.
+    choices: [ 'yes', 'no', 'ask', 'auto', 'autoask' ]
+    type: str
```
Comment (felixfontein) [inline plugins/modules/ssh_config.py:115]: Same for the other new options.
```suggestion
    type: str
    version_added: 8.1.0
```

**Review by felixfontein — COMMENTED** [review]
Thanks for your contribution! I've added a first comment. You might also want to extend the integration tests (in tests/integration/targets/ssh_config/) to test these new options.

File: plugins/modules/ssh_config.py (hunk 1 above)
Comment (mering) [reply plugins/modules/ssh_config.py:115]: Done.

**Comment by mering** [comment]
Thanks for the review! I added these options to the integration test.

File: plugins/modules/ssh_config.py [hunk 2]
```diff
@@ -214,19 +246,17 @@ def ensure_state(self):
             port=self.params.get('port'),
             identity_file=self.params.get('identity_file'),
             user=self.params.get('remote_user'),
-            strict_host_key_checking=self.params.get('strict_host_key_checking'),
+            strict_host_key_checking=fix_bool_str(self.params.get('strict_host_key_checking')),
```
Comment (felixfontein) [inline plugins/modules/ssh_config.py:249]: Why is this needed? `strict_host_key_checking` can only take the three string values `'yes'`, `'no'`, and `'ask'`. It can never be `'True'` or `'False'`.

File: plugins/modules/ssh_config.py [hunk 3]
```diff
@@ -214,19 +246,17 @@ def ensure_state(self):
...
             user_known_hosts_file=self.params.get('user_known_hosts_file'),
             proxycommand=self.params.get('proxycommand'),
             proxyjump=self.params.get('proxyjump'),
             host_key_algorithms=self.params.get('host_key_algorithms'),
+            forward_agent=convert_bool(self.params.get('forward_agent')),
+            controlmaster=fix_bool_str(self.params.get('controlmaster')),
```
Comment (felixfontein) [inline plugins/modules/ssh_config.py:255]: The same is true for `controlmaster`.

File: plugins/modules/ssh_config.py (hunk 2 above)
Comment (mering) [reply plugins/modules/ssh_config.py:249]: When setting this value in the YAML playbook to `yes` instead of `"yes"`, YAML interprets this as boolean, Python will use `bool` (`True`) type, the module class converts it to `str` (`"True"`) as per `type` specification. While this is a user error of using the wrong data type, it is very common and difficult to spot.

Comment (felixfontein) [reply plugins/modules/ssh_config.py:249]: `strict_host_key_checking` has `choices` set (https://github.com/ansible-collections/community.general/blob/4df626dd5d3e293e9219f05d1aaafd428ee9a025/plugins/modules/ssh_config.py#L355), so `AnsibleModule` will make sure that its value is exactly one of these strings.

The argspec validator actually has special code to handle converting `'False'` and `'True'` back to 'equivalent' choices strings (in this case: `yes` and `no`): https://github.com/ansible/ansible/blob/devel/lib/ansible/module_utils/common/parameters.py#L672-L683 If it wouldn't have this code, the module would fail if you provide a boolean here (since `'True'` and `'False'` are not on the choices list).

Comment (felixfontein) [reply plugins/modules/ssh_config.py:249]: The second commit should not be dropped without a replacement since `controlpersist` can use such a conversion. Since it has `type='str'` without `choices` you end up with `'True'` and `'False'` when passing booleans.

**Review by felixfontein — APPROVED** [review]
Looks good to me! If nobody objects, I'll merge in ~a week.

**Comment by felixfontein** [comment]
@mering thanks for improving the module!
